### PR TITLE
Add systemd unit file for insights-client-boot service

### PR DESCRIPTION
**Pull request by aleccohan**
Based on conversations with @subpop.  This PR will add a systemd unit file that will _upon declaration of a file run_insights_client_next_boot_ will run the insights client on the next boot of the host node.  This will be triggered by playbooks created by remediations that are specified with the `localhost` tag.

See Corresponding Remediations JIRA:  https://github.com/RedHatInsights/insights-remediations/pull/12

### CODE REVIEW

**Comment by aleccohan** [comment]
@subpop @gravitypriest If you have the time today would you be able to take a look at this?  Thanks!

**Comment by aleccohan** [comment]
@subpop @gravitypriest could you take a look at this PR?  Thanks.

File: data/systemd/insights-client-boot.service [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ExecStart=/usr/bin/insights-client --retry 3
+Restart=no
+WatchdogSec=900
+CPUQuota=30%
+MemoryLimit=2G
+TasksMax=300
+BlockIOWeight=100
+ExecStartPost=-/bin/bash -c "echo 2G >/dev/null 2>&1 > /sys/fs/cgroup/memory/system.slice/insights-client.service/memory.memsw.limit_in_bytes"
+ExecStartPost=-/bin/bash -c "echo 1G >/dev/null 2>&1 > /sys/fs/cgroup/memory/system.slice/insights-client.service/memory.soft_limit_in_bytes"
+ExecStartPost=-rm /etc/insights-client/.run_insights_client_next_boot
```
Comment (Ichimonji10) [inline data/systemd/insights-client-boot.service:27]: Why ignore return codes on these three commands?

For this final command,  adding the `--force` flag will cause rm to "ignore nonexistent files and arguments, never prompt."

**Review by gravitypriest — APPROVED** [review]

File: data/systemd/insights-client-boot.service [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+After=network.target
+ConditionPathExists=/etc/insights-client/.run_insights_client_next_boot
+
+[Service]
+Type=oneshot
+ExecStart=/usr/bin/insights-client --retry 3
+Restart=no
+WatchdogSec=900
+CPUQuota=30%
+MemoryLimit=2G
```
Comment (Ichimonji10) [inline data/systemd/insights-client-boot.service:22]: `MemoryMax=` replaces `MemoryLimit=`, no need for both.

**Comment by Ichimonji10** [comment]
Thanks for revising!

File: data/systemd/insights-client-boot.service (hunk 2 above)
Comment (gravitypriest) [reply data/systemd/insights-client-boot.service:22]: This also needs to work in RHEL 7 -- I'm not sure whether you can just use `MemoryLimit` as an alias or whether you can define both, but we should go for max compatibility here.

Comment (Ichimonji10) [reply data/systemd/insights-client-boot.service]: Gotcha. :+1:

**Review by gravitypriest — APPROVED** [review]

**Pull request merged**